### ISSUE TITLE
A.2.6 Federated catalogues advertisement

### ISSUE THREAD

**Issue by jensscheerlinck** [inspire-eu-validation/ats-discovery-service#14]
_This issue has been extracted from the issue list on:https://ies-svn.jrc.ec.europa.eu/issues/2685_
# Comment

The validation can only be done knowing the Discovery Scenario chosen when registering the service in the INSPIRE Geoportal.

**Comment by jensscheerlinck** [comment]
# Proposed Resolution

Agreed with comment. This test is not testable in an automated way.
The problem lies indeed with determining whether a Discovery Service was registered as federated or not in the INSPIRE Geoportal.
Once this has been established, test A.2.7 covers the rest.

XPath: "/ows:OperationsMetadata/ows:Constraint[@name='FederatedCatalogues']"

**Comment by PeterParslow** [comment]
Agreed: this requirement is conditional on a condition that cannot be automatically checked.

**Comment by PeterParslow** [comment]
Although I don't believe the condition is to do with registering in the INSPIRE geoportal. From TG Requirement 3, and section 4.3.4, the condition is whether the service under test (Member State Discovery Service) has chosen the Discovery Client Approach or the Discover Service (federated) approach to implementing Link Discovery Service - AND that subsidiary services have registered themselves. If the service under test has chosen the Centralised approach, then Requirement 3 doesn't apply - even if there are subsidiary discovery services, they are not 'federated'.
The only way that information is available to the INSPIRE Geoportal is by satisfying Requirement 3.

**Comment by aquaglia** [comment]
As of today, the additional information about the Discovery Scenario is only available in the service registry of the geoportal.

Implementation Requirement 3 only says that "The list of federated catalogues, if any, shall be advertised as the result of a Service metadata response to a Discover Metadata request."
However, every Discovery Service must be described in a metadata document regardless of the scenario.

Third Party Discovery Services can additionally be published in the “FederatedCatalogues” section of the Discovery Service’s capability document if they are part of a federated search infrastructure.

So, it does not say that the list is empty when the Scenario is the CLIENT one.

Historically, the Service scenario has consistently failed because of the high response times and of fragile implementations of caching mechanisms in existing software.

Poland, Austria and Italy are currently using the CLIENT scenario.
The first two have the additional need to limit the harvesting to only some of the Discovery Services documented in metadata.
If the FederatedCatalogues element is empty all linked discovery services are harvested.
If it is not empty (except for the top discovery service), only the listed services are harvested.

**Comment by PeterParslow** [comment]
UK adopted the centralised scenario. Regional & domain catalogues have metadata records, which get passed on to the EU. They register to be harvested (a separate process, though it needn't be). So no catalogues are federated & therefore Requirement 3 doesn't apply.
